includes/Admin/Admin_AJAX.php

```diff
@@ -46,6 +50,100 @@ public function get_nonce() {
 		return wp_create_nonce( self::NONCE_KEY );
 	}
 
+	/**
+	 * Handles the AJAX request to setup the runtime environment if needed.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function setup_environment() {
+		// Verify the nonce before continuing.
+		$valid_nonce = $this->verify_nonce( filter_input( INPUT_POST, 'nonce', FILTER_SANITIZE_STRING ) );
+
+		if ( is_wp_error( $valid_nonce ) ) {
+			wp_send_json_error( $valid_nonce, 403 );
+		}
+		$runner = Plugin_Request_Utility::get_runner();
+
+		if ( is_null( $runner ) ) {
+			$runner = new AJAX_Runner();
+		}
+
+		// Make sure we are using the correct runner instance.
+		if ( ! ( $runner instanceof AJAX_Runner ) ) {
+			wp_send_json_error(
+				new WP_Error( 'invalid-runner', __( 'AJAX Runner was not initialized correctly.', 'plugin-check' ) ),
+				500
+			);
+		}
+
+		$checks = wp_parse_list( filter_input( INPUT_POST, 'checks', FILTER_SANITIZE_STRING ) );
```

Comment:
See my comment in https://github.com/10up/plugin-check/pull/100#discussion_r1142610721, why is this not an array of strings?

Reply:
@felixarntz I responded on https://github.com/10up/plugin-check/pull/100#discussion_r1143636914 but to clarify I have found a solution to ensuring an array is passed in the JS `FormData`. This required appending each checks to the array so it is passed to the AJAX endpoint as expected.

I've also updated the `AJAX_Runner` and unit tests which were not setup correctly.